### ISSUE TITLE
Monitoring mechanism checking commands for model modifications

### ISSUE THREAD

**Issue by MWEclipse** [MWEclipse/testing#155]
| --- | --- |
| Bugzilla Link | [528690](https://bugs.eclipse.org/bugs/show_bug.cgi?id=528690) |
| Status | ASSIGNED |
| Importance | P3 normal |
| Reported | Dec 13, 2017 04:07 EDT |
| Modified | Aug 20, 2023 14:14 EDT |
| Version | 1.15.0 |
| See also | [Gerrit change https://git.eclipse.org/r/113305](https://git.eclipse.org/r/113305), [Gerrit change https://git.eclipse.org/r/113641](https://git.eclipse.org/r/113641), [Git commit 63141405](https://git.eclipse.org/c/emfclient/org.eclipse.emf.ecp.core.git/commit/?id=631414054ae1512fcb150af2290938af7e177814) |
| Reporter | Remi Schnekenburger |

## Description 
For existing applications, it may be interesting to check if all model modifications are run through commands, to enable undo / redo for example.

It can be interesting for that purpose to provide a Transactional editing domain that checks the presence of command in the stack trace when a write operation is done on the model.\
This should be optional, and it should not impact users that do not opt for monitoring.

A log should be generated for each model modification not run through a command. In order to reduce log size, a unique stack should be logged only once per session.

**Comment by MWEclipse** [comment]
Dec 13, 2017 04:09

New Gerrit change created: https://git.eclipse.org/r/113305

**Comment by MWEclipse** [comment]
Dec 20, 2017 03:34

Gerrit change https://git.eclipse.org/r/113641 was merged to [feature_528690_commandmonitoring].\
Commit: http://git.eclipse.org/c/emfclient/org.eclipse.emf.ecp.core.git/commit/?id=631414054ae1512fcb150af2290938af7e177814

**Comment by MWEclipse** [comment]
By Eugen Neufeld on Feb 13, 2018 05:32

What is the status here?

**Comment by MWEclipse** [comment]
By Remi Schnekenburger on Mar 02, 2018 07:22

Bug has been postponed to next target milestone, until the monitoring framework has been used.

**Comment by MWEclipse** [comment]
By Eugen Neufeld on May 16, 2018 05:04

Will this be merged to develop before the RC1? Otherwise move it to the 1.18.0 Milestone. Thank you.

**Comment by MWEclipse** [comment]
By Eugen Neufeld on Aug 22, 2018 09:35

Mass Move due to time constraints

**Comment by MWEclipse** [comment]
By Remi Schnekenburger on Nov 14, 2018 07:23

Postponing again to next target milestone.

**Comment by MWEclipse** [comment]
By Remi Schnekenburger on May 28, 2019 04:39

postponing again to next target milestone.

**Comment by MWEclipse** [comment]
Aug 20, 2023 14:14

This bug hasn't had any activity in quite some time. Maybe the problem got resolved, was a duplicate of something else, or became less pressing for some reason - or maybe it's still relevant but just hasn't been looked at yet.

If you have further information on the current state of the bug, please add it. The information can be, for example, that the problem still occurs, that you still want the feature, that more information is needed, or that the bug is (for whatever reason) no longer relevant.

--\
The automated Eclipse Genie.
